manager/manager.go

```diff
@@ -127,11 +128,13 @@ func (ps *ProbeSelector) RunValidator(manager *Manager) error {
 	if !ok {
 		return errors.Errorf("probe not found: %s", ps.ProbeIdentificationPair)
 	}
-	if !p.IsRunning() && p.Enabled && !p.Optional {
+	if !p.IsRunning() && p.Enabled {
 		return errors.Wrap(p.GetLastError(), ps.ProbeIdentificationPair.String())
 	}
-	if !p.Enabled && !p.Optional {
-		return errors.Errorf("%s: is disabled and not optional", ps.ProbeIdentificationPair.String())
+	if !p.Enabled {
+		return errors.Errorf(
```

Comment:
It appears that explicitly disabled probes return errors from the validator. Is the assumption that this validator will not be called?

Reply:
Exactly. The `ActivatedProbes` selector should only contain probes that will be activated, or expect at least one success in a list of failures (which is why we introduced `OneOf`).

Also if you look at how the list of selectors is generated in https://github.com/DataDog/datadog-agent/pull/6371, you'll see that only the `enabledProbes` are added.

Reply:
So my answer is not really answering your question.

In order to declare what probes should be enabled, you need to add your probes in the `ActivatedProbes`. If your probe was successfully activated (= not excluded by the `ExcludedSections` parameter + the kernel was actually successful in activating your probe), a `ProbeSelector` will successfully validate the activation.

If for whatever reason a probe ends up not being enabled (which might happen if its section is excluded), then the `ProbeSelector` fill fail the validation. But, if you put this `ProbeSelector` within a `OneOf` validator, there is still a chance that the overall validation succeeds, because of the other probes within the `OneOf` validator.

So the last reason why a `ProbeSelector` might fail the validation is if the kernel refused the probe, in which case the validation process will yield an error, including the error from the kernel.

I hope that makes sense.